### PR TITLE
BING-25 creates user model

### PR DESCRIPTION
**Pull request by vytautel**
Creates user model for MongoDB

### CODE REVIEW

File: app/bingey-api/models/Users.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const { Schema, model } = require('mongoose');
+
+const UsersSchema = new Schema({
```
Comment (palianskas) [inline app/bingey-api/models/Users.js]: We should add timestamping to this schema

File: app/bingey-api/models/Users.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    username: String,
+    password: String,
+    watchlistLink: String,
```
Comment (palianskas) [inline app/bingey-api/models/Users.js]: The schema should follow the specified db model and `required` should be added to username and password (not watchlist probably)

File: app/bingey-api/models/Users.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+});
+
+const Users = model('Users', UsersSchema);
```
Comment (palianskas) [inline app/bingey-api/models/Users.js]: The model name should be singular, as it will be pluralized when establishing a connection with the collection

**Review by palianskas — CHANGES_REQUESTED** [review]

**Review by MatasRut — CHANGES_REQUESTED** [review]
I created a preliminary db structure file, you can find it in #important discord chat so you should follow those guidelines. Also watchlistLink should be called watchlistId, should be of type Schema.Types.ObjectId and reference the Watchlist schema. You can pass an object instead of the property type like so: { type: Schema.Types.ObjectId, ref: 'Watchlist', required: true }

File: app/bingey-api/models/user.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+const { Schema, model } = require('mongoose');
+
+const userSchema = new Schema({
+    _id:{
```
Comment (palianskas) [inline app/bingey-api/models/user.js]: User should have autogenerated IDs, the _Title_ had manually added ones cause we will use the ones from IMDb

File: app/bingey-api/models/user.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                ref: 'Watchlist',
+            },
+        }
+    ],
+    
+	recommended: [
+		{
+            titleId: {
+                type: Schema.Types.ObjectId,
+                ref: 'Titles',
```
Comment (palianskas) [inline app/bingey-api/models/user.js]: Mongoose has backward-compatibility breaking quirk with pluralization
```suggestion
                ref: 'Title',
```

File: app/bingey-api/models/user.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                type: String,
+            },
+			imgUrl: {
+                type: String,
+            },
+		}
+	]
+},
+  {
+    _id: false,
```
Comment (palianskas) [inline app/bingey-api/models/user.js]: This shouldn't be added as autogenerated IDs should be used

File: app/bingey-api/models/user.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+	recommended: [
+		{
+            titleId: {
+                type: Schema.Types.ObjectId,
+                ref: 'Titles',
+            },
+            name: {
+                type: String,
+            },
+			imgUrl: {
```
Comment (palianskas) [inline app/bingey-api/models/user.js:38]: This has some strange indentation haha

File: app/bingey-api/models/user.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    
+	recommended: [
+		{
+            titleId: {
+                type: Schema.Types.ObjectId,
+                ref: 'Titles',
+            },
+            name: {
+                type: String,
+            },
```
Comment (palianskas) [inline app/bingey-api/models/user.js]: I think keeping  such field declarations ( _fieldName: type_ ) shorter would be better
```suggestion
            name: String,
```

**Review by palianskas — CHANGES_REQUESTED** [review]

**Review by palianskas — CHANGES_REQUESTED** [review]
The indentation is all over the place, does the Prettier not solve it for some reason?
Also, it's not necessary right now as we don't have any actual authentication, but the _password_ field could have a `select: false` option so as to not return it in queries

**Review by palianskas — APPROVED** [review]

**Pull request merged**